profiler/src/test/java/com/splunk/opentelemetry/profiler/snapshot/SpecialTraceIds.java

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.opentelemetry.profiler.snapshot;
+
+import io.opentelemetry.api.trace.TraceId;
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Queue;
+import java.util.Random;
+
+/**
+ * Maintains a collection of trace IDs mapped to percentiles matching the probability calculation in
+ * {@link io.opentelemetry.sdk.trace.samplers.TraceIdRatioBasedSampler}.
+ *
+ * <p>A trace ID percentile is defined as the range of values existing between {@link
+ * Long#MAX_VALUE} * (percentile / 100) and {@link Long#MAX_VALUE} * (percentile + 1 / 100)
+ *
+ * <pre>
+ * Ex. For percentile 5 the range is considered to be:
+ * lower = {@link Long#MAX_VALUE} * 0.04 = 368934881474191040
+ * upper = {@link Long#MAX_VALUE} * 0.05 = 461168601842738816
+ * </pre>
+ *
+ * <p>A trace ID is considered to be in the percentile range when the absolute value of its hash
+ * value (as calculated using {@link
+ * io.opentelemetry.api.internal.OtelEncodingUtils#longFromBase16String(traceId, 16)}) is within the
+ * range.
+ *
+ * @see io.opentelemetry.sdk.trace.samplers.TraceIdRatioBasedSampler
+ * @see io.opentelemetry.api.internal.OtelEncodingUtils#longFromBase16String(CharSequence, int)
+ */
+class SpecialTraceIds {
+  /**
+   * Get trace IDs known to be within the requested percentile. At least two traces will be returned
+   * for each percentile representing both positive and negative hash value calculations.
```

Comment:
> representing both positive and negative hash value calculations

perhaps rephrase to `representing second long in the trace id being both positive and negative`